pallets/loans/src/functions.rs

```diff
@@ -245,12 +248,39 @@ impl<T: Config> Pallet<T> {
 
 				Self::update_nav_with_updated_present_value(pool_id, new_pv, old_pv)
 			},
-		)?;
+		)
+		.map(|(count, _)| count)
+	}
+
+	pub(crate) fn extend_loan(
+		pool_id: PoolIdOf<T>,
+		loan_id: T::LoanId,
+		added_time: Moment,
+	) -> Result<ActiveCount, DispatchError> {
+		Self::try_mutate_active_loan(
+			pool_id,
+			loan_id,
+			|active_loan| -> Result<(), DispatchError> {
+				let new_maturity_date = active_loan
+					.loan
+					.schedule
+					.maturity_date()
+					.ensure_add(added_time)?;
+
+				let now = Self::now();
+
+				ensure!(new_maturity_date > now, Error::<T>::LoanMaturityDatePassed);
+
+				active_loan
+					.loan
+					.schedule
+					.update_maturity_date(new_maturity_date);
 
-		Ok((
-			ActiveLoans::<T>::get(pool_id).len().try_into().unwrap(),
-			interest_rate_per_sec,
-		))
+				// TODO: should update PV of loan
```

Comment:
Should this be the same implementation as found in `reprice_loan()` method?

Reply:
Yes.